components/Schedule.js

```diff
@@ -0,0 +1,170 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import Link from "next/link";
+
+const StyledDiv = styled.div`
+  text-align: center;
+`;
+const StyledImg = styled.img`
+  border-radius: 5px;
+  width: 100px;
+  height: 100px;
+  margin-right: 10px;
+`;
+const StyledList = styled.ul`
+  list-style: none;
+  padding: 0;
+`;
+const StyledItem = styled.li`
+  display: block;
+  align-items: left;
+  align-content: space-between;
+  margin-bottom: 5px;
+  list-style: none;
+  border-spacing: 10px;
+`;
+const StyledInfo = styled.div`
+  display: flex;
+  flex-direction: column;
+  text-align: left;
+  margin-right: 5px;
+`;
+const StyledPlant = styled.div`
+  display: flex;
+  position: relative;
+  margin-bottom: 5px;
+  color: grey;
+  border-style: solid;
+  padding: 10px;
+  margin: 10px;
+`;
+const StyledName = styled.p`
+  margin-right: 5px;
+  font-weight: bold;
+  color: black;
+`;
+const StyledSpan = styled.span`
+  margin-right: 1rem;
+`;
+const StyledLink = styled(Link)`
+  text-decoration: none;
+`;
+const SuccessMessage = styled.p`
+  display: inline-block;
+  width: 90%;
+  color: green;
+  background-color: lightgreen;
+  border: 3px solid green;
+  border-radius: 0.5rem;
+  padding: 0.5rem;
+`;
+const AddSchedule = styled.p`
+  display: inline-block;
+  border: 3px solid var(--secondary-stroke-color);
+  background-color: var(--secondary-bg-color);
+  margin-top: 2rem;
+  padding: 0.8rem 0.8rem;
+  border-radius: 1rem;
+  color: white;
+  text-decoration: none;
+  font-weight: bold;
+  cursor: pointer;
+  &:hover {
+    color: var(--secondary-stroke-color);
+    background-color: white;
+  }
+`;
+export default function MySchedule({
+  plants,
+  isMyPlantWithReminder,
+  plantsInfo,
+}) {
+  const router = useRouter();
+  const [successMessage, setSuccessMessage] = useState("");
+
+  useEffect(() => {
+    if (router.query.success) {
+      setSuccessMessage("Watering-Schedule added successfully!");
+    } else {
+      setSuccessMessage("");
+    }
+  }, [router.query.success]);
+
+  const plantsWithReminder = plants
+    .filter((plant) => isMyPlantWithReminder(plant.id))
+    .map((plant) => {
+      const plantInfo = plantsInfo?.find((info) => info.id === plant.id);
+      return {
+        ...plant,
+        wateringDate: plantInfo?.wateringDate || null,
+      };
+    })
+    .sort((a, b) => new Date(a.wateringDate) - new Date(b.wateringDate));
```

Comment:
please use toSorted instead of sort, since it does not mutate the previous array.